python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -57,11 +59,13 @@ def validate_resource(client, key, resource, actions):
     return errors
 
 
-def get_subnet(client, subnet, subnet_parts):
+def get_subnet(cli_ctx, subnet, subnet_parts):
     try:
-        subnet_obj = client.subnets.get(subnet_parts['resource_group'],
-                                        subnet_parts['name'],
-                                        subnet_parts['child_name_1'])
+        subnet_obj = subnet_show(cli_ctx=cli_ctx)(command_args={
+            'resource_group': subnet_parts['resource_group'],
+            'name': subnet_parts['child_name_1'],
+            'vnet_name': subnet_parts['name']
+        })
     except ResourceNotFoundError as err:
         raise InvalidArgumentValueError(
             f"Invalid -- subnet, error when getting '{subnet}': {str(err)}") from err
```

Comment:
@ellis-johnson when refactoring this code I noticed this error message looked like it was missing `--{key.replace('_', '-')}`, is that right?  If so I could work that into this PR but I think it will need some additional changes to work with `dyn_validate_cidr_ranges`, or just pass None as key value since this exception shouldn't be raised in `dyn_validate_cidr_ranges` as it would be caught when validating subnets.

Reply:
Woops I didn't realise that was the intention of `--` in these error messages so that was just a bad copy on my behalf. I'm happy to just pass None for the error in `dyn_validate_cidr_ranges`. Or even to remove the `--` in the message itself.

Reply:
Done!  Passed key as None so this aligns with the rest of the code.  Thanks for checking this!